src/transforms/transform-code-scanning-alert.ts

```diff
@@ -101,3 +109,43 @@ export const transformCodeScanningAlert = async (context: WebhookContext, github
 		}]
 	};
 };
+
+export const transformCodeScanningAlertToJiraSecurity = async (context: WebhookContext, githubInstallationId: number, jiraHost: string): Promise<JiraVulnerabilityBulkSubmitData | null> => {
+	const { alert, repository } = context.payload;
+
+	if (!alert.most_recent_instance?.ref?.startsWith("refs/heads")) {
+		context.log.info("Skipping code scanning alert detected on a pull request.");
+		return null;
+	}
+
+	const metrics = {
+		trigger: "webhook",
+		subTrigger: "code_scanning_alert"
+	};
+	const gitHubInstallationClient = await createInstallationClient(githubInstallationId, jiraHost, metrics, context.log, context.gitHubAppConfig?.gitHubAppId);
+
+	const handleUnmapped = (state) => context.log.info(`Received unmapped state from code_scanning_alert webhook: ${state}`);
+
+	return {
+		vulnerabilities: [{
+			schemaVersion: "1.0",
+			id: `c-${transformRepositoryId(repository.id, gitHubInstallationClient.baseUrl)}-${alert.number}`,
+			updateSequenceNumber: Date.now(),
+			containerId: transformRepositoryId(repository.id, gitHubInstallationClient.baseUrl),
+			displayName: alert.rule.name,
+			description: alert.rule.full_description || alert.rule.description,
+			url: alert.html_url,
+			type: "sast",
+			introducedDate: alert.created_at,
+			lastUpdated: alert.dismissed_at || alert.fixed_at || alert.updated_at || alert.created_at,
+			severity: {
+				level: transformGitHubSeverityToJiraSeverity(alert.rule.security_severity_level, handleUnmapped)
+			},
+			identifiers: [],
```

Comment:
This field is actually required. I didn't specify it in the RFC since I wasn't sure yet what value to use here. 

We will need to explore the payload to see what values we can use here, there is an `alert.rule.tags` but doesn't seem to have the URL so we might need to build this ourselves. There is a URL in `alert.rule.help` but I'm not sure if that is available for all action types and backfilling so will need to check.

Reply:
Added the transformation of `alert.rule.tags` to vulnerability `identifiers`. When no CWE tags are found, `identifiers` will be omitted.